src/components/TextInputFocusable/index.js

```diff
@@ -182,6 +189,42 @@ class TextInputFocusable extends React.Component {
         return newNumberOfLines;
     }
 
+    /**
+     * Handles all types of drag-N-drop events on the composer
+     *
+     * @param {Object} e native Event
+     * @memberof TextInputFocusable
+     */
+    dragNDropListener(e) {
+        let isOriginComposer = false;
+        const handler = () => {
+            switch (e.type) {
+                case 'dragover':
+                    e.preventDefault();
+                    this.props.onDragOver(e, isOriginComposer);
+                    break;
+                case 'dragenter':
+                    e.dataTransfer.dropEffect = 'copy';
+                    this.props.onDragEnter(e, isOriginComposer);
+                    break;
+                case 'dragleave': this.props.onDragLeave(e, isOriginComposer); break;
+                case 'drop': this.props.onDrop(e, isOriginComposer); break;
```

Comment:
I'm not sure if we have a convention yet, but I think we should just keep these on separate lines. Unless each case was a one liner, I think this is hard to read
```
case 0: return func.call(this, rest);
case 1: return func.call(this, arguments[0], rest);
case 2: return func.call(this, arguments[0], arguments[1], rest);
```